tests/operator/helper.go

```diff
@@ -45,3 +52,87 @@ func SplitCsv(csv string) CsvResult {
 	}
 	return result
 }
+
+func hasOperatorInstallModeType(installModes []v1alpha1.InstallMode, installModeType v1alpha1.InstallModeType) bool {
+	for i := 0; i < len(installModes); i++ {
+		if installModes[i].Type == installModeType && installModes[i].Supported {
+			return true
+		}
+	}
+	return false
+}
+
+func getAllPodsBy(namespace string) (podsInNamespace []*provider.Pod) {
+	for i := range env.AllPods {
+		pod := env.AllPods[i]
+		if pod.Namespace == namespace {
+			podsInNamespace = append(podsInNamespace, pod)
+		}
+	}
+	return podsInNamespace
+}
+
+func checkValidOperatorPodsIn(namespace string, allPods []*provider.Pod) (foundCsvs map[string]bool, podsBelongingToNoOperators []string, err error) {
+	foundCsvs = make(map[string]bool)
+	for index := range allPods {
+		pod := allPods[index]
+		topOwners, err := podhelper.GetPodTopOwner(pod.Namespace, pod.OwnerReferences)
+		if err != nil {
+			return foundCsvs, podsBelongingToNoOperators, err
+		}
+
+		validOwnerFound := false
+		for _, owner := range topOwners {
+			if owner.Kind == v1alpha1.ClusterServiceVersionKind && owner.Namespace == namespace {
+				foundCsvs[owner.Name] = true
+				validOwnerFound = true
+				break
+			}
+		}
+		if !validOwnerFound {
+			podsBelongingToNoOperators = append(podsBelongingToNoOperators, pod.Name)
+		}
+	}
+	return foundCsvs, podsBelongingToNoOperators, nil
+}
+
+// This function checks if the namespace contains only valid operator pods
+func checkIfNamespaceContainsOnlyOperatorPods(namespace string) (isOperatorOnlyNamespace bool, singleNamespacedCsvs, allNamespacedCsvs, podsBelongingToNoOperators []string, err error) {
+	allPods := getAllPodsBy(namespace)
+
+	foundCsvs, podsBelongingToNoOperators, err := checkValidOperatorPodsIn(namespace, allPods)
+	if err != nil {
+		return false, singleNamespacedCsvs, allNamespacedCsvs, podsBelongingToNoOperators, err
+	}
+
+	allOperatorsFoundInNamespaceAreValid := true
+
+	for _, operator := range env.Operators {
+		for foundCsv := range foundCsvs {
+			if operator.Csv.Name != foundCsv {
+				continue
+			}
+
+			// Handle cluster-wide operators
+			if operator.IsClusterWide {
+				allOperatorsFoundInNamespaceAreValid = false
+				break
+			}
+
+			if hasOperatorInstallModeType(operator.Csv.Spec.InstallModes, v1alpha1.InstallModeTypeAllNamespaces) {
+				allOperatorsFoundInNamespaceAreValid = false
+				allNamespacedCsvs = append(allNamespacedCsvs, foundCsv)
+				break
+			}
+			if hasOperatorInstallModeType(operator.Csv.Spec.InstallModes, v1alpha1.InstallModeTypeSingleNamespace) {
```

Comment:
Same here. There's no need to check the supported modes. It was installed in own/singleNamespace mode if its operatorgroup's `len(spec.targetNamespaces) == 1`.